addons/mail/static/src/models/messaging_notification_handler/messaging_notification_handler.js

```diff
@@ -692,30 +692,29 @@ function factory(dependencies) {
         _notifyNewChannelMessageWhileOutOfFocus({ channel, message }) {
             const author = message.author;
             const messaging = this.env.messaging;
-            let notificationTitle;
+            let notificationHeader;
             if (!author) {
-                notificationTitle = this.env._t("New message");
+                notificationHeader = this.env._t("New message");
             } else {
                 const authorName = author.nameOrDisplayName;
                 if (channel.channel_type === 'channel') {
-                    const groupIcon = '<i class="fa fa-hashtag"/> ';
-                    const privateIcon = '<i class="fa fa-lock"/> ';
-                    const channelIcon = channel.public === 'private' ? privateIcon : groupIcon;
+                    const channelIcon = channel.public === 'private' ? '<i class="fa fa-lock"/> '
+                                                                     : '<i class="fa fa-hashtag"/> ';
                     const channelName = owl.utils.escape(channel.displayName);
                     const channelNameWithIcon = channelIcon + channelName;
-                    notificationTitle = _.str.sprintf(
-                        this.env._t("<b>%s from %s</b>"),
+                    notificationHeader = _.str.sprintf(
+                        this.env._t("%s from %s"),
                         owl.utils.escape(authorName),
                         channelNameWithIcon
                     );
                 } else {
-                    notificationTitle = _.str.sprintf(
-                        this.env._t("<b>%s</b>"),
-                        owl.utils.escape(authorName)
-                    );
+                    notificationHeader = owl.utils.escape(authorName);
                 }
             }
-            const notificationContent = notificationTitle + message.prettyBody.substr(0, PREVIEW_MSG_MAX_SIZE);
+            document.body.classList.add('o_notification_discuss');
+            const notificationContent = _.str.sprintf(
+                                        this.env._t("<b>%s</b></br>"),
+                                        notificationHeader) + mailUtils.htmlToTextContentInline(message.prettyBody);
```

Comment:
In case the message is too long `substr` should still be kept, but on the text content (non html).